allennlp/training/metrics/metric.py

```diff
@@ -34,6 +38,26 @@ def get_metric(
         """
         raise NotImplementedError
 
+    def get_metric_name_value_pairs(
+        self, default_name: str, reset: bool = False
+    ) -> Iterable[Tuple[str, float]]:
+        """
+        Return the metric as in `self.get_metric` but as an iterable of string-float pairs.
+        """
+        value = self.get_metric(reset)
+        if isinstance(value, collections.abc.Mapping):
+            for sub_name, sub_value in value.items():
+                if isinstance(sub_value, collections.abc.Iterable):
+                    for i, sub_value_i in enumerate(sub_value):
+                        yield f"{sub_name}_{i}", sub_value_i
+                else:
+                    yield sub_name, sub_value
+        elif isinstance(value, collections.abc.Iterable):
+            for i, sub_value in enumerate(value):  # type: ignore
```

Comment:
mypy associates the previous variable (`sub_value`) with the same name the same type as this one, so I added this ignore.